ddl/rollingback.go

```diff
@@ -117,6 +118,51 @@ func rollingbackAddColumn(t *meta.Meta, job *model.Job) (ver int64, err error) {
 	return ver, errCancelledDDLJob
 }
 
+func rollingbackDropIndex(t *meta.Meta, job *model.Job) (ver int64, err error) {
+	var indexName model.CIStr
+	err = job.DecodeArgs(&indexName)
+	if err != nil {
```

Comment:
If an index's state is `StateWriteOnly` and the index enters the current logic, then this job is finished.
Is there a problem with this situation?

Reply:
Great catch. I will fix it.

Reply:
@crazycs520 
I can't understand why you said: "the index state will also be public". The index's state is `StateWriteOnly`, and this job returns in this logic(this job will be cancel), then the index's state still is `StateWriteOnly`.

Reply:
I am just giving an example here. Or is it possible to get an error in `getTableInfo`?

Reply:
ye， if `getTableInfo` return error, we should canceled job here.

Reply:
In `getTableInfo`, the job's state is set as `model.JobStateCancelled`

Reply:
In `getTableInfo`,  the job's state is set as model.JobStateCancelled only when the error is `ErrDBNotExists` or `ErrTableNotExists` or `tblInfo.State != model.StatePublic`. In this case, we should canceled `drop index` ddl job.